test/es6/dynamic-module-import-syntax.js

```diff
@@ -0,0 +1,83 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// ES6 Generators Syntax tests -- verifies function* and yield syntax spec conformance
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+function testScript(source, message, shouldFail = false, explicitAsync = false) {
+    message += " (script)";
+    let testfunc = () => testRunner.LoadScript(source, undefined, shouldFail, explicitAsync);
+
+    if (shouldFail) {
+        let caught = false;
+
+        assert.throws(testfunc, SyntaxErrr, message);
+        assert.isTrue(caught, `Expected error not thrown: ${message}`);
+    } else {
+        assert.doesNotThrow(testfunc, message);
+    }
+}
+
+function testModuleScript(source, message, shouldFail = false, explicitAsync = false) {
+    message += " (module)";
+    let testfunc = () => testRunner.LoadModule(source, 'samethread', shouldFail, explicitAsync);
+
+    if (shouldFail) {
+        let caught = false;
+
+        // We can't use assert.throws here because the SyntaxError used to construct the thrown error
+        // is from a different context so it won't be strictly equal to our SyntaxError.
+        try {
+            testfunc();
+        } catch(e) {
+            caught = true;
+
+            // Compare toString output of SyntaxError and other context SyntaxError constructor.
+            assert.areEqual(e.constructor.toString(), SyntaxError.toString(), message);
+        }
+
+        assert.isTrue(caught, `Expected error not thrown: ${message}`);
+    } else {
+        assert.doesNotThrow(testfunc, message);
+    }
+}
+
+var tests = [
+    {
+        name: "Valid cases for import()",
+        body: function () {
+                let functionBody =
+                    `
+                    assert.doesNotThrow(function () { eval("import(undefined)"); }, "undefined");
```

Comment:
They are not exactly the same. Import-specifier.js tests ToString operation applied on specifier. This one tests parser.

---
In reply to: [115364519](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115364519) [](ancestors = 115364519)